src/test/java/org/owasp/validator/html/test/AntiSamyTest.java

```diff
@@ -1509,5 +1509,21 @@ public void testGithubIssue99() throws ScanException, PolicyException {
         assertThat(as.scan("<p lang=\"en-GB\">This paragraph is defined as British English.</p>", policy, AntiSamy.DOM).getCleanHTML(), containsString("lang=\"en-GB\""));
         assertThat(as.scan("<p lang=\"en-GB\">This paragraph is defined as British English.</p>", policy, AntiSamy.SAX).getCleanHTML(), containsString("lang=\"en-GB\""));
     }
+
+    @Test
+    public void testGithubIssue101() throws ScanException, PolicyException {
+        // Test that margin attribute is not removed when value has too much significant figures.
+        // Current behavior is that decimals like 0.0001 are internally translated to 1.0-E4, this
+        // is reflected on regex validation and actual output. The inconsistency is due to Batik CSS.
+        assertThat(as.scan("<p style=\"margin: 0.0001pt;\">Some text.</p>", policy, AntiSamy.DOM).getCleanHTML(), containsString("margin"));
+        assertThat(as.scan("<p style=\"margin: 0.0001pt;\">Some text.</p>", policy, AntiSamy.SAX).getCleanHTML(), containsString("margin"));
+        assertThat(as.scan("<p style=\"margin: 10000000pt;\">Some text.</p>", policy, AntiSamy.DOM).getCleanHTML(), containsString("margin"));
+        assertThat(as.scan("<p style=\"margin: 10000000pt;\">Some text.</p>", policy, AntiSamy.SAX).getCleanHTML(), containsString("margin"));
+        // When using exponential directly the "e" or "E" is internally considered as the start of
+        // the dimension/unit type. This creates inconsistencies that make the regex validation fail,
+        // also in cases like 1e4pt where "e" is considered as dimension instead of "pt".
+        assertThat(as.scan("<p style=\"margin: 1.0E+4pt;\">Some text.</p>", policy, AntiSamy.DOM).getCleanHTML(), not(containsString("margin")));
```

Comment:
IIRC, the GitHub issue that originally was brought up was when someone was trying to use a fractional point, something like 0.001pt or whatever. This tests for a margin of 10000pt which seems a bit excessive. (Had Fermat had that much room in his margins, "Fermat's Last Theorem" likely would have never been a thing! :)
Anyway, my point is, there should be at least something here with a negative exponent to test the original use case that was causing the problem.

Reply:
I get the point, 10000pt was just to test the behavior of a positive exponential (also 0.0001 is excesively low for a margin in my opinion). Test was introduced.